test/tests/typeguards.test.ts

```diff
@@ -0,0 +1,43 @@
+import { expect } from 'chai';
+import { isDocument, isDocumentArray } from '../../src/utils';
+import { UserRefModel } from '../models/userRefs';
+
+export function typeguards() {
+    it ('should guarantee array of document types', async () => {
+        const UserMaster = await UserRefModel.create({
+            name: 'master'
+        });
+        const UserSub = await UserRefModel.create({
+            name: 'sub',
+            master: UserMaster._id
+        });
+        UserMaster.subAccounts.push(UserSub._id);
```

Comment:
```suggestion
        UserMaster.subAccounts.push(UserSub);
```
It works without `._id` and is more clean

Reply:
in practice this almost never happens, you have the ref and you want to explicitly populate, it's this way in order to work as example too